app/src/main/java/com/github/geohunt/app/data/repository/ActiveHuntsRepository.kt

```diff
@@ -47,30 +47,51 @@ class ActiveHuntsRepository(
      *
      * This adds the challenge to his active hunts page.
      */
-    suspend fun joinHunt(challenge: Challenge) = updateHuntState(challenge, doHunt = true)
+    override suspend fun joinHunt(challenge: Challenge) = updateHuntState(challenge, doHunt = true)
 
     /**
      * Makes the currently authenticated user leave the hunt on the given challenge.
      *
      * This removes the challenge to his active hunts page.
      */
-    suspend fun leaveHunt(challenge: Challenge) = updateHuntState(challenge, doHunt = false)
+    override suspend fun leaveHunt(challenge: Challenge) = updateHuntState(challenge, doHunt = false)
 
 
     /**
      * Returns the active hunts IDs of the currently authenticated user.
      *
      * If there is no currently authenticated user, throws a [UserNotLoggedInException].
      */
-    fun getActiveHunts(): Flow<List<String>> {
+    override fun getActiveHunts(): Flow<List<String>> {
         authRepository.requireLoggedIn()
 
         val currentUser = authRepository.getCurrentUser()
 
+        // List<String> should not be stored as is within the database,
+        // prefer Map<String, Boolean>
         return activeHunts
             .child(currentUser.id)
             .snapshots
-            .map { it.toList() }
+            .map {
+                it.children.mapNotNull { dataSnapshot ->
+                    dataSnapshot.key.takeIf { dataSnapshot.getValue(Boolean::class.java) ?: false }
+                }
+            }
+            .flowOn(ioDispatcher)
+    }
+
+    /**
+     * Check whether or not the currently authenticated user hunt a specific challenges
+     */
+    override fun getDoesHunts(challenge: Challenge) : Flow<Boolean> {
```

Comment:
This naming is confusing, maybe something like `isHunting` would be clearer ?
```suggestion
    override fun isHunting(challenge: Challenge) : Flow<Boolean> {
```